model/project_event.go

```diff
@@ -108,12 +108,35 @@ func (p *ProjectChangeEvents) RedactPrivateVars() {
 		if !isChangeEvent {
 			continue
 		}
+		modifiedPrivateVarKeys := findModifiedPrivateVars(changeEvent)
 		changeEvent.After.Vars = *changeEvent.After.Vars.RedactPrivateVars()
 		changeEvent.Before.Vars = *changeEvent.Before.Vars.RedactPrivateVars()
+		// Here we need to change the after value of modified private variables to
+		// {REDACTED} so the event log will show that the private variable
+		// has been changed in the UI. This is necessary because the above RedactPrivateVars
+		// function redacts private variables to an empty string, which will not get picked up
+		// by the UI as it relies on the before / after diff.
+		for _, privateVarKey := range modifiedPrivateVarKeys {
+			changeEvent.After.Vars.Vars[privateVarKey] = "{REDACTED}"
```

Comment:
Just to verify after reading the other threads + the new changes - it's not possible to un-private a variable via the REST route, correct? You can't un-private variable via Spruce, but I wonder if the REST route lets you do more things that would let you change its privacy. My concern here would only be that if that's possible, we should also redact the before value as well, in case someone un-privates a variable unintentionally via the REST route.

Reply:
It's not possible to un-private a variable via the REST route, correct.

The reason that only the before or after value shows {REDACTED} is because the UI picks up on the diffs between the before / after event, so if both the before and after value are {REDACTED} nothing will show up in the UI. I changed this to be {REDACTED_BEFORE} -> {REDACTED_AFTER} to be a bit clearer